app/upgrades/v2_5_0/v2_5_0.go

```diff
@@ -53,145 +51,44 @@ var Upgrade = upgrades.Upgrade{
 	StoreUpgrades: storetypes.StoreUpgrades{},
 }
 
-// Set of addresses that held stNIBI in ERC20 form prior to the v2.5.0 upgrade
-func MAINNET_STNIBI_HOLDERS() []gethcommon.Address {
-	holderStrs := set.New(
-		"0x603871c2ddd41c26Ee77495E2E31e6De7f9957e0",
-		"0x447fC0471f5d8150A790e19000930108aCCe0BC7",
-		"0x2DD37531749e1AF248720fe8F9D1A51517D9748F",
-		"0x07A1D54115D5B535F298231Cc6EaA360Bc4667f9",
-		"0x5361EBC7CF6689565EC4C380b7793e238ad6Be45",
-	)
-	holders := make([]gethcommon.Address, len(holderStrs))
-	for idx, h := range holderStrs.ToSlice() {
-		holders[idx] = gethcommon.HexToAddress(h)
-	}
-	return holders
-}
-
-var (
-	// MAINNET_STNIBI_ADDR is the (real) hex address of stNIBI on mainnet.
-	MAINNET_STNIBI_ADDR = gethcommon.HexToAddress("0xcA0a9Fb5FBF692fa12fD13c0A900EC56Bb3f0a7b")
-
-	// MAINNET_NIBIRU_SAFE_ADDR: Address of a Gnosis Safe managed by the Nibiru team
-	// See https://nibiscan.io/address/0x22CBd7CbF3b33681abB3Ced4D64d71acB9a9dCd2/contract/6900/code
-	MAINNET_NIBIRU_SAFE_ADDR = gethcommon.HexToAddress("0x22CBd7CbF3b33681abB3Ced4D64d71acB9a9dCd2")
-)
+// MAINNET_STNIBI_ADDR is the (real) hex address of stNIBI on mainnet.
+var MAINNET_STNIBI_ADDR = gethcommon.HexToAddress("0xcA0a9Fb5FBF692fa12fD13c0A900EC56Bb3f0a7b")
 
 func UpgradeStNibiContractOnMainnet(
-	nibiru *keepers.PublicKeepers,
+	keepers *keepers.PublicKeepers,
 	ctx sdk.Context,
 	// erc20Addr is the hex address of stNIBI on mainnet
 	// The upgrade handler takes the address as an argument for testing purposes
-	erc20Addr gethcommon.Address,
+	originalErc20Addr gethcommon.Address,
 ) error {
 	// -------------------------------------------------------------------------
 	// STEP 0: Early return if the FunToken mapping or ERC20 has an invalid state
 	// -------------------------------------------------------------------------
 
 	// Early return if the mapping for stNIBI doesn't exist
-	erc20AddrIter := nibiru.EvmKeeper.FunTokens.Indexes.ERC20Addr.ExactMatch(ctx, erc20Addr)
-	funTokenMappings := nibiru.EvmKeeper.FunTokens.Collect(ctx, erc20AddrIter)
+	erc20AddrIter := keepers.EvmKeeper.FunTokens.Indexes.ERC20Addr.ExactMatch(ctx, originalErc20Addr)
+	funTokenMappings := keepers.EvmKeeper.FunTokens.Collect(ctx, erc20AddrIter)
 	if len(funTokenMappings) != 1 {
 		return nil
 	}
 
 	// Early return if the address is not a contract
-	accOfContract := nibiru.EvmKeeper.GetAccount(ctx, erc20Addr)
-	if accOfContract == nil || !accOfContract.IsContract() {
+	originalErc20Account := keepers.EvmKeeper.GetAccount(ctx, originalErc20Addr)
+	if originalErc20Account == nil || !originalErc20Account.IsContract() {
 		return nil
 	}
 
-	// -------------------------------------------------------------------------
```

Comment:
As explained above, we no longer need to query the original token balances anymore.